### ISSUE TITLE
Better documentation for graphql_scalar

### ISSUE THREAD

**Issue by nalply** [graphql-rust/juniper#515]
**Is your feature request related to a problem? Please describe.**
I need a more complex scalar implementation because of validation. Validation is an application-wide concern, and I would like to implement some more features not related to GraphQL.

**Describe the solution you'd like**
The graphql_scalar macro is very complex, but neither Rust docs nor the Juniper book explain to what exactly the graphql_scalar expands. I resorted to reading the macro, and I already found out that it expands to an `impl <...> GraphQLType ScalarName { ... }`. It should not be neccessary to read source, especially a very complex macro definition.

**Describe alternatives you've considered**
Reading the macro and/or generating macro source with `cargo expand`.

**Additional context**
Perhaps there are other areas of the library needing some love?

**Comment by davidpdrsn** [comment]
If you managed to solve your problem how do you feel about submitting a PR with improvements? I have personally not used the scalar type macro before so I wouldn't really know what users are missing.

**Comment by nalply** [comment]
I agree, Juniper could need help improving the documentation. Just today I stumbled over IntoResolvable, and when I looked at the API doc it was nowhere to found, so looked at the source and discovered this offending line:

https://github.com/graphql-rust/juniper/blob/824f329b0f23718fc996437aad76361a82e46453/juniper/src/executor/mod.rs#L235

I'll keep Juniper in mind as something where I really could help out, but let me dive even deeper into Juniper first. It's a very exciting and important project.
